Patches/GameOptionsMenuPatch.cs

```diff
@@ -718,52 +712,238 @@ public static void Postfix(RoleSettingsTabMenu __instance)
-        __instance.ImpostorLightMod = 1.75f;
-        __instance.KillCooldown = 27.5f;
-        __instance.NumCommonTasks = 2;
-        __instance.NumLongTasks = 1;
-        __instance.NumShortTasks = 2;
-        __instance.NumEmergencyMeetings = 3;
-        if (!isOnline)
-            __instance.NumImpostors = NormalGameOptionsV08.RecommendedImpostors[numPlayers];
-        __instance.KillDistance = 0;
-        __instance.DiscussionTime = 0;
-        __instance.VotingTime = 120;
-        __instance.IsDefaults = true;
-        __instance.ConfirmImpostor = false;
-        __instance.VisualTasks = false;
-
-        __instance.roleOptions.SetRoleRate(RoleTypes.Shapeshifter, 0, 0);
-        __instance.roleOptions.SetRoleRate(RoleTypes.Scientist, 0, 0);
-        __instance.roleOptions.SetRoleRate(RoleTypes.GuardianAngel, 0, 0);
-        __instance.roleOptions.SetRoleRate(RoleTypes.Engineer, 0, 0);
-        __instance.roleOptions.SetRoleRecommended(RoleTypes.Shapeshifter);
-        __instance.roleOptions.SetRoleRecommended(RoleTypes.Scientist);
-        __instance.roleOptions.SetRoleRecommended(RoleTypes.GuardianAngel);
-        __instance.roleOptions.SetRoleRecommended(RoleTypes.Engineer);
-
-        if (Options.CurrentGameMode == CustomGameMode.FFA) //FFA
+        __instance.selectedRoleTab = 0;
+        __instance.RoleChancesSettings.SetActive(true);
+
+        _ = new LateTask(() =>
         {
-            __instance.CrewLightMod = __instance.ImpostorLightMod = 1.25f;
-            __instance.NumImpostors = 3;
-            __instance.NumCommonTasks = 0;
-            __instance.NumLongTasks = 0;
-            __instance.NumShortTasks = 0;
-            __instance.KillCooldown = 0f;
-        }
+            LoadRoleOptions(__instance, VanillaLikeRoleTypes.Crewmate);
+        }, 0.1f, "OpenFirstChancesTab");
 
         return false;
     }
-}
-*/
\ No newline at end of file
+
+    private static void CreateCustomQuotaOption(RolesSettingsMenu inst, RoleBase dog, ref float yPos, int index)
+    {
+        RoleOptionSetting roleOptionSetting = Object.Instantiate(inst.roleOptionSettingOrigin, Vector3.zero, Quaternion.identity, inst.RoleChancesSettings.transform);
+        roleOptionSetting.transform.localPosition = new Vector3(-0.15f, yPos, -2f);
+
+        // Set Role
+        SpriteRenderer[] componentsInChildren = roleOptionSetting.GetComponentsInChildren<SpriteRenderer>(true);
+        for (int i = 0; i < componentsInChildren.Length; i++)
+        {
+            componentsInChildren[i].material.SetInt(PlayerMaterial.MaskLayer, 20);
+        }
+        foreach (TextMeshPro textMeshPro in roleOptionSetting.GetComponentsInChildren<TextMeshPro>(true))
+        {
+            textMeshPro.fontMaterial.SetFloat("_StencilComp", 3f);
+            textMeshPro.fontMaterial.SetFloat("_Stencil", 20);
+        }
+        roleOptionSetting.role = RoleManager.Instance.AllRoles[0];
+        roleOptionSetting.titleText.text = Utils.GetRoleName(dog.ThisRoleBase);
```

Comment:
Use here GetKeyByValue https://github.com/0xDrMoe/TownofHost-Enhanced/commit/dc6785cee05b3a1031178ccbe8c75a516aa9c25c
Using `var customRole = CustomRoleManager.RoleClass.GetKeyByValue(dog);`

Reply:
btw why rolebase named as dog? lol

Reply:
U can also just do `Enum.Parse<CustomRoles>(dog.GetType().name, true);`

Reply:
btw is there a reason for taking it by normal options to get the rolebase list, instead of simply doing e. g  `var roles = Enum.GetValues<CustomRoles>().Where(r => !r.Isaditionalrole() && r.isCrewmate())` ?